src/wfs/src/test/java/org/geoserver/wfs/WFSReprojectionUtilTest.java

```diff
@@ -4,19 +4,64 @@
  */
 package org.geoserver.wfs;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertNull;
 
+import org.geotools.data.DataUtilities;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.referencing.CRS;
+import org.geotools.util.factory.GeoTools;
 import org.junit.Test;
 import org.opengis.feature.type.FeatureType;
+import org.opengis.filter.Filter;
+import org.opengis.filter.FilterFactory2;
+import org.opengis.filter.spatial.BBOX;
+import org.opengis.referencing.FactoryException;
 import org.opengis.referencing.crs.CoordinateReferenceSystem;
 
 public class WFSReprojectionUtilTest {
 
+    private final FeatureType featureType = createFeatureType();
+    private final FilterFactory2 filterFactory = createFilterFactory();
+
+    private static FeatureType createFeatureType() {
+        try {
+            return DataUtilities.createType(
+                    "testType", "geom:Point:srid=4326,line:LineString,name:String,id:int");
+        } catch (Exception exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    private static FilterFactory2 createFilterFactory() {
+        try {
+            return CommonFactoryFinder.getFilterFactory2(GeoTools.getDefaultHints());
+        } catch (Exception exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
     @Test
     public void testNullCheck() {
         // used to throw an NPE if the feature type was null
         CoordinateReferenceSystem crs =
                 WFSReprojectionUtil.getDeclaredCrs((FeatureType) null, "1.1.0");
         assertNull(crs);
     }
+
+    @Test
+    public void testReprojectFilterWithTargetCrs() throws FactoryException {
```

Comment:
This looks good, but would it make sense to also have a test covering app-schema?
Axis flipping wise, would be interesting to see WFS 1.0 vs WFS 2.0, but I'm guessing one cannot do app-schema with WFS 1.0 (all app-schemas being based on GML3... or not?)

Reply:
You are correct, I should have added a specific App-Schema test for this, doing it. Regarding GML2, well in theory it shoudl work ... but we don't have any test that uses GML2 and I never saw an use case actually using GML2, so I will let it aside.